### ISSUE TITLE
NavState and SE_2(3)

### ISSUE THREAD

**Issue by dellaert** [borglab/gtsam#1937]
## Feature

We want NavState, a manifold, to also get Lie group operations so it becomes SE2(3).

Note: NavState was a perfectly fine manifold, which sometimes is all you need. We could add a class SE_2(3), but we will follow the current GTSAM convention and add the group operations in the same class.

## Motivation

Many new VIO filters and smoothers now use SE2(3) for improved consistency.

## TODO:

Each of these should be a separate PR.

- [x] Basic Group operations, done in #1930 
- [x] Add Functor to streamline Jacobians, done in #1932 
- [x] Simplify Computation of Q blocks, done in #1938 
- [x] Investigate nearZero behavior and selection of thresholds - chose an ok threshold
- [ ] Fix coriolis terms according to Brossard22tro
- [ ] transpose group operations, unit tested
- [ ] Switch retract and localCoordinates to exp/log. If we remove the custom `retract` and `localCoordinates`, lots of unit tests fail, so it is important to check this too.
- [ ] Use in "Brossard's code"

**Comment by dellaert** [comment]
FYI @varunagrawal @ProfFan @stefangachter 
Let me still play with this for a bit, no action need from y'all except review PRs.

**Comment by dellaert** [comment]
codex: This tracks adding Lie-group operations to `NavState` so it can represent SE_2(3), plus the remaining follow-up work. Labeled enhancement because it expands the mathematical API and capabilities.

**Comment by dellaert** [comment]
The #1916 work has been split into two focused replacements:\n\n- #2687 — NavState Lie-group IMU preintegration\n- #2688 — exact rotating-Earth IMU/AHRS prediction (currently a draft stacked on #2687)\n\n#2688 implements the Brossard transition centrally for all preintegration backends and factors, while preserving covariance propagation and archive compatibility. The Coriolis / Brossard checklist items should be updated only after #2688 merges.
